### PR TITLE
#185: Created a C++ file and updated the README file for Reversing a Doubly Linked List Problem

### PR DESCRIPTION
**Pull request by Pragati1122**
## Related Issue
- #185 
- The problem is to reverse a doubly linked list such that the flow of the doubly linked list reverses and the last node node becomes head while the head node becomes the last node.
- Added the code for Reverse a Doubly Linked List problem by using the method of swapping the values for the previous and next pointers of each node.
- Added the explanation for Reverse a Doubly Linked List problem in the README.md file.

## Proposed Changes
- Added the C++ file for Reverse a Doubly Linked List problem.
- Added README.md file of Reverse a Doubly Linked List problem with time and space complexity.

## Additional Info
- This code is a simple method for reversing a doubly linked list by the method of swapping previous and next pointers for all nodes, changing previous of the head (or start) and change the head pointer in the end.
- This is one of the most common questions from the topic of Doubly Linked Lists.
- This code solves an open issue Reverse a Doubly Linked List  #185
- I have tested these changes on VS code using multiple inputs and matched the output with the expected output.
- This change will not affect or create conflict with any other code
- Time Complexity : O(n) where n is the number of nodes in the linked list.
- Space Complexity: O(1)

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| README preview  | Code Snippet     |
| image  | image    |

### CODE REVIEW

**Review by miraj0507 — COMMENTED** [review]
This looks good to me, kindly change the push(&head,6) style, and create a user input style.

**Comment by n4i9kita** [comment]
@Pragati1122  please make the changes as requested by @miraj0507

File: L-LinkedList/D-DoublyLinkedList/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Linked List:
+A linked list is a linear data structure, in which the elements are not stored at contiguous memory locations. The elements in a linked list are linked using pointers as shown in the image below:
+
+![ll](https://media.geeksforgeeks.org/wp-content/cdn-uploads/gq/2013/03/Linkedlist.png)
+
+In simple words, a linked list consists of nodes where each node contains a data field and a reference(link) to the next node in the list. 
+
+## Why Linked List ?
```
Comment (n4i9kita) [inline L-LinkedList/D-DoublyLinkedList/README.md:8]: I think it is unrelated to the problem.

File: L-LinkedList/D-DoublyLinkedList/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+2. Change prev of the head (or start).
+3. Change the head pointer in the end.
+
+## Time Complexity:
+Time Complexity : O(n), where N denotes the number of nodes in the doubly linked list.
+
+## Space Complexity:
+Space Complexity : O(1), (since no extra space is used.)
+
+## Algorithm
```
Comment (n4i9kita) [inline L-LinkedList/D-DoublyLinkedList/README.md:64]: These are statements, and not an algorithm. Please refer to the readme.md in the `S-Sorting` Folder for reference.

Also, avoid writing information that is not relevant to the algorithm you are implementing, or the problem statement you are solving.

**Comment by Pragati1122** [comment]
> This looks good to me, kindly change the push(&head,6) style, and create a user input style.

Can you please elaborate? I could not understand.

**Comment by n4i9kita** [comment]
@Pragati1122 it means that rather than yourself initializing the linked list in the main(), take the data from the user that you want to store in the linked list.

**Pull request closed without merging**